esmf-semantic-aspect-meta-model/src/main/resources/samm/meta-model/2.2.0/aspect-meta-model-shapes.ttl

```diff
@@ -64,7 +64,7 @@ samm:NamespaceShape
          select $this ?value ?code
          where {
             bind( str( $this ) as ?value )
-            filter( !regex( ?value, "^urn:samm:[^:]+:[0-9]+.[0-9]+.[0-9]+#$" ))
+            filter( !regex( ?value, "^urn:samm:[a-zA-Z0-9._-]+:[0-9]+.[0-9]+.[0-9]+#$" ))
```

Comment:
I think this would still allow actually invalid names such as "00...aaa". We should build the pattern up following the rules we intent to enforce:
* The pattern consists of parts which are separated by dots
* Since we're using "reverse domain name notation" and each domain name has at least a name and a TLD, there must be at least two parts
* The first part, representing the TLD, must be a valid TLD, and therefore follow RFC 952: Must be at least 2 characters, `a-zA-Z0-9` allowed but no number at the beginning
* The second part, representing the domain's name, has a a minimum length of only 1, but also allows `-`. Leading numbers are allowed
* All following parts, if there are any, allow `a-zA-Z0-9_-`
* Every part has a max length of 63 (RFC 2128 section 11)

So my proposal would be:
```suggestion
            filter( !regex( ?value, "^urn:samm:[a-zA-Z][a-zA-Z0-9]{1,62}\\.[a-zA-Z0-9-]{1,63}(\\.[a-zA-Z0-9_-]{1,63})*:[0-9]+.[0-9]+.[0-9]+#$" ))
```

Note that the `.` character needs to be doubly escaped, once for in the regex and then the regex-escaping-`\` needs to be escaped in the RDF/Turtle.